### ISSUE TITLE
Add graph to tensorboard fails

### ISSUE THREAD

**Issue by Mateusz-Grzelinski** [BindsNET/bindsnet#494]
`bindsnet.network.Network` does not implement `forward` method, what causes TensorBoard `add_graph` to fail. Is there any reason why bindsnet does not use this function? Sorry if this is silly question, but I am quite new to this package and I spent 2 days figuring it out.

See minimal example:

```python
import torch
from bindsnet.analysis.pipeline_analysis import TensorboardAnalyzer
from bindsnet.network import Network
from bindsnet.network.nodes import Input


class ValidTensorboardNetwork(Network):
    def forward(self, *input):
        print(input)
        return input

# network = Network() # fails
network = ValidTensorboardNetwork() # works

input_layer = Input(shape=(1, 1), traces=True)

dummy_input = (torch.zeros(1, 1),)

analyzer = TensorboardAnalyzer("logs/demo")
analyzer.writer.add_graph(network, dummy_input, verbose=True)
```

From `torch.utils.tensorboard.writer.SummaryWriter`:
```python
    def add_graph(self, model, input_to_model=None, verbose=False):
        if hasattr(model, 'forward'):
            # A valid PyTorch model should have a 'forward' method
            self._get_file_writer().add_graph(graph(model, input_to_model, verbose))
```

**Comment by Hananel-Hazan** [comment]
Thank you for using BindsNET. 

BindsNET dont use the forward method in the traditional sense since BindsNET dont use gradient descent, instead BindsNET use the `run` method to propagate the information in the layers. 
However, you can use the main features of tensorboard with BindsNET, take a look on the tensorboard [example](https://github.com/BindsNET/bindsnet/blob/master/examples/tensorboard/tensorboard.py)
